packages/next/src/lib/find-root.ts

```diff
@@ -23,21 +23,24 @@ export function findRootDir(cwd: string) {
 
   const lockFiles = [lockFile]
   while (true) {
-    const nextDir = dirname(dirname(lockFiles[lockFiles.length - 1]))
-    const newLockFile = findRootLockFile(nextDir)
+    const lastLockFile = lockFiles[lockFiles.length - 1]
+    const currentDir = dirname(lastLockFile)
+    const parentDir = dirname(currentDir)
 
-    if (newLockFile) {
-      lockFiles.push(newLockFile)
-    } else {
-      break
-    }
+    if (parentDir === currentDir) break
```

Comment:
```suggestion
    // dirname('/')==='/' so if we happen to reach the FS root (as might happen in a container we need to quit to avoid looping forever
    if (parentDir === currentDir) break
```